### DISCUSSION TITLE
Easier curation of Global Feed on Nodes

### DISCUSSION THREAD

**Discussion by arhebbar** [deso-protocol/dips#150] [Medium CIPs]
# One-Line Description
Automatic Whitelisting of content from Whitelisted Profiles or Option in Admin to do so.

# Explanation and Motivation
Currently, I believe this happens at the time of posting, not at the time of content curation and that makes it very painful for any new node to curate good content on their feed and gives BitClout.com a huge advantage because most of the content will continue to get posted on BitClout.com and will automatically get whitelisted on the global feed. 

I should be able to say, I like @Krassenstein 's content. Let 5 of his most popular posts go Global every day. Not have to manually go click on 5 of his profiles even after adding him/her to the Grantlist.

After having managed the content curation on feed.creatiers.co for just a week, I can assure you that this is critical to nodes surviving and thriving and in turn BitClout platform surviving and thriving. 

# Specification
This will require updates to the core, backend and frontend (including admin) to fully implement.

I would like this CIP to focus on the core & backend changes to enable this functionality on the frontend and nodes can take care of the settings to see how they use it.

# Data Storage Change
None

# Backwards Compatibility
This is only about future posts going global and not the past. Hence, there should be no change beyond more posts going global on more nodes and hence more engagement on the platform.

**Comment by tijno** [comment]
agree this would be great

how would it be implemented though ? eg when will it check for posts from whitelisted users and add them to global?

**Reply by arhebbar** [reply]
> how would it be implemented though ? eg when will it check for posts from whitelisted users and add them to global?

The way I would do this is that I would whitelist a post for all users depending on whether it is among the Top 5 posts of that user for the day or whatever the current check is to whitelist a post. When showing the Global Feed on a given node, I'd filter for those in the Grant List for that Node and show only their whitelisted posts. Any additional explicitly whitelisted posts also can get added to the Global feed. 

e.g. If you are not grantlisted and still post something that I as an Admin feel it belongs to Global on my node, I will whitelist that post. So, my node's global feed will include up to 5 posts of all grantlisted users and explicitly whitelisted posts of other users.

**Reply by tijno** [reply]
yeah i get that - but practically when would this "adding 5 to global" happen ?

right now it can happen on the node the user posts from in the process triggered by them posting.

but to solve the problem you outlined - that is not happening.

so something needs to trigger the process of identifying the posts that should be in global and creating the index.

i dont think this should be done live when people load global

it will be too slow.

I also dont think it can be done when the block is received as that would require changes to core but core is really about blockchain concensus.

**Reply by arhebbar** [reply]
I am honestly not the expert on the current code and might need clarity on what all happens when a block is received today before I can chime in. 

But, my best guess from a logic perspective would be that when a block is received would be the right time and cadence to index the posts from the new block that belong to global, along with any other indexes that get rebuilt once new data is received.

**Comment by krassenstein** [comment]
Noticed I was tagged in this and I'm not totally sure why.

Might have meant to tag a different Krassenstein. I'm the game developer one. Sorry!

**Reply by arhebbar** [reply]
Sorry ... yes, a different krassenstein.

**Comment by stet** [comment]
pay users to post on your node. use _PostExtraData_ as needed. 
ideally, this is the feature that should be worked on for core (to avoid trusted manual processes/scripts/bots).
then your users will have incentive to use your new node and the rest may follow.

but yes, this is an issue and just thinking about the solution has given me headaches in past.
